Packs/EmailCommunication/Scripts/SummarizeEmailThreads/SummarizeEmailThreads.py

```diff
@@ -0,0 +1,114 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def fetch_email_threads(incident_id):
+    """Fetch all Email Threads stored in the current incident context
+    Args:
+        incident_id (str): The incident ID
+    Returns:
+        list of dicts. List of email thread entries
+    """
+    # Get current email threads from context if any are present
+    try:
+        incident_context = demisto.executeCommand("getContext", {'id': incident_id})
+        threads = dict_safe_get(incident_context[0], ['Contents', 'context', 'EmailThreads'])
+
+        if not threads:
+            return None
+
+        # Return a list of dicts, even if there is only one thread entry and the context item is a list
+        threads = [threads] if isinstance(threads, dict) else threads
+
+        return threads
+
+    except Exception as e:
+        print(f'Exception in fetch_email_threads: {e}')
+
+
+def format_threads(email_threads):
+    """Process list of email thread context entries and return a table in MD format
+    Args:
+        email_threads (list of dicts): List of email thread entry dictionaries
+    Returns:
+        md (str): Markdown formatted table
+    """
+    try:
+        thread_summary_md = []
+        for thread_entry in email_threads:
+            # Loop through thread entries.  'EmailCommsThreadNumber' is indexed from 0 so we can use it for list indexing
+            # Trim off thread ID code for readability
+            thread_number = thread_entry['EmailCommsThreadNumber']
```

Comment:
if this is needed in n int form in from this place on, please do it here, rather than turning it to int many times later in the code.

Reply:
Done.